src/Generator/IdGenerator.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\MediaBundle\Generator;
+
+use Sonata\MediaBundle\Model\MediaInterface;
+
+/**
+ * @final since sonata-project/media-bundle 3.21.0
+ */
+class IdGenerator implements GeneratorInterface
+{
+    /**
+     * @var int
+     */
+    protected $firstLevel;
+
+    /**
+     * @var int
+     */
+    protected $secondLevel;
+
+    /**
+     * @param int $firstLevel
+     * @param int $secondLevel
+     */
+    public function __construct($firstLevel = 100000, $secondLevel = 1000)
```

Comment:
You can use type declarations rightaway

Reply:
And remove the phpdoc as it does not add any value

Reply:
But then also `DefaultGenerator` class will be using typed declarations.
Isn't this a BC break?

Reply:
Ah you are correct, if we use a class aliases, you can't do this. I hadn't that in mind when writing this comment.

Reply:
even if you use extension, in fact.